lib/board.rb

```diff
@@ -4,49 +4,52 @@ class Board
   PLAYER_ONE = 1
   PLAYER_TWO = 2
 
-  attr_reader :errors, :positions
+  attr_reader :error, :positions
 
   def initialize(positions = EMPTY_BOARD)
-    @errors = []
+    @error = nil
     @positions = positions
   end
 
   def available_positions
-    available = []
-    @positions.each_with_index { |pos, idx| available << -pos * (idx + 1) }
-    available.reject { |pos| pos < 0 }
+    @positions.map.with_index { |position, index| index if position < 0 }.reject(&:nil?)
   end
 
   def place_token(position, player)
     position_valid?(position) ? apply_token(position, player) : apply_errors(position)
   end
 
-  def has_errors?
-    @errors.length > 0
+  def has_error?
+    @error != nil
   end
 
   private
 
   def position_valid?(position)
-    is_integer?(position) && available_positions.include?(position)
+    is_integer?(position) && available_positions.include?(position.to_i)
   end
 
   def validation_reason(position)
-    !is_integer?(position) || position < 1 || position > @positions.length ? :invalid_position : :position_taken
+    !is_integer?(position) || is_outside_board_range?(position) ? :invalid_position : :position_taken
   end
 
   def is_integer?(position)
     Integer(position).is_a? Integer rescue false
```

Comment:
```
rescue false
```

🙀 this feels like it's overly-aggressively catching all errors. might want to try a less potentially destructive method of checking input, like a Regular Expression.

Reply:
Thanks @bppr that's useful. I've followed your advice and updated the method.